lib/iterator.js

```diff
@@ -199,6 +199,67 @@ class Iterator {
     return result + suffix;
   }
 
+  // Find value in this iterator compared by `comparator` on values
+  // obtained by applying `accessor` to each value.
+  // Signature: comparator, [accessor[, thisArg]]
+  //   comparator <Function> returns true if new value should be accepted.
+  //     currValue <any> current value, starts with undefined
+  //     nextValue <any> next value
+  //     Returns: <boolean> true if next value should be accepted.
+  //   accessor <Function> gets value to compare by, current iterator value
+  //       is used by default.
+  //     value <any> this
+  //     Returns: <any> value to compare by.
+  //   thisArg <any> value to be used as `this` when calling `accessor` and
+  //       `comparator`.
+  // Returns: element with maximum `accessor` value.
+  findCompare(comparator, accessor, thisArg) {
+    let res = undefined;
+    let resCompareBy = undefined;
+    for (const value of this) {
+      const compareBy = accessor ? accessor.call(thisArg, value) : value;
+      if (comparator.call(thisArg, resCompareBy, compareBy)) {
+        resCompareBy = compareBy;
+        res = value;
+      }
+    }
+    return res;
+  }
+
+  // Find maximum value in this iterator compared by values obtained by
+  // applying `accessor` to each value.
+  // Signature: [accessor[, thisArg]]
+  //   accessor <Function> gets value to compare by, current iterator value
+  //       is used by default.
+  //     value <any> this
+  //     Returns: <any> value to compare by.
+  //   thisArg <any> value to be used as `this` when calling `accessor`.
+  // Returns: element with maximum `accessor` (iterator value by default) value.
```

Comment:
It's unclear that 'iterator value by default' refers to the fact that if a user doesn't pass `accessor` function, values itself will be used for comparison. Apart from that, it's not actually `accessor` value that is being compared since it's the same for every element and is equal to the function that was passed into this method, so you should rewrite this to make it more clear. You also need to specify that it can return undefined for an empty iterator.